daphne_worker/src/config.rs

```diff
@@ -264,10 +264,10 @@ impl<D> DaphneWorkerConfig<D> {
 
     // Get a reference to the HPKE receiver configs, ensuring that the config indicated by
     // `hpke_config_id` is cached (if it exists).
-    pub(crate) async fn cached_hpke_receiver_configs(
-        &self,
+    pub(crate) async fn hpke_receiver_config<'a>(
+        &'a self,
```

Comment:
I think the explicit lifetimes here could be elided turning the signature of this method into the following instead with no changes to functionality
```rust
pub(crate) async fn hpke_receiver_config(
        &self,
        hpke_config_id: u8,
    ) -> Result<Option<GuardedHpkeReceiverConfig>> {
```

Reply:
Holy smokes, you're right! I was under the impression that, since it's part of the type, we'd have to name the parameter here. Maybe we don't have to name it in the trait? Let me give it a shot...

Reply:
I've applied this change and played with ways we might remove the explicit lifetime from the traits. Still not clear if there's a way to do this.